File: python/struse.py
# (c) 2026 : https://github.com/NMoroney + Gemini CLI
# MIT License
# https://github.com/NMoroney/Struse
# https://nmoroney.github.io/notes/2026/260423_struse/index.html
#

import random

m1 = {
    "a": "&#x10300;",  # OLD ITALIC LETTER A 
    "b": "&#x1041A;",  # DESERET CAPITAL LETTER VEE 
    "c": "&#x1103;",   # HANGUL CHOSEONG TIKEUT
    "d": "&#x1095;",   # MYANMAR SHAN DIGIT FIVE
    "e": "&#x0F5B;",   # TIBETAN LETTER DZA
    "f": "&#x0CEF;",   # KANNADA DIGIT NINE
    "g": "&#x0D6D;",   # MALAYALAM DIGIT SEVEN
    "h": "&#x0A2E;",   # GURMUKHI LETTER MA
    "i": "&#x1109;",   # HANGUL CHOSEONG SIOS
    "j": "&#x10408;",  # DESERET CAPITAL LETTER SHORT A 
    "k": "&#x051E;",   # CYRILLIC CAPITAL LETTER ALEUT KA 
    "l": "&#x10539;",  # CAUCASIAN ALBANIAN LETTER CHA 
    "m": "&#x1328;",   # ETHIOPIC SYLLABLE CHA
    "n": "&#x0E17;",   # THAI CHARACTER THO THAHAN 
    "o": "&#x07DB;",   # NKO LETTER SA
    "p": "&#x10502;",  # ELBASAN LETTER CE
    "q": "&#x09ED;",   # BENGALI DIGIT SEVEN
    "r": "&#x053B;",   # ARMENIAN CAPITAL LETTER INI 
    "s": "&#x0D17;",   # MALAYALAM LETTER GA 
    "t": "&#x10B5;",   # GEORGIAN CAPITAL LETTER KHAR
    "u": "&#x053F;",   # ARMENIAN CAPITAL LETTER KEN
    "v": "&#x13C9;",   # CHEROKEE LETTER QUO
    "w": "&#x0E1F;",   # THAI CHARACTER FO FAN 
    "x": "&#x0514;",   # CYRILLIC CAPITAL LETTER LHA
    "y": "&#x080A;",   # SAMARITAN LETTER KAAF
    "z": "&#x10350;"   # OLD PERMIC LETTER AN
}

m2 = {
    "a": "&#x1D70;",   # LATIN SMALL LETTER N WITH MIDDLE TILDE
    "b": "&#x266D;",   # MUSIC FLAT SIGN 
    "c": "&#x122D;",   # ETHIOPIC SYLLABLE RE
    "d": "&#x1D1CE;",  # MUSICAL SYMBOL TEMPUS IMPERFECTUM CUM PROLATIONE IMPERFECTA DIMINUTION-3
    "e": "&#x096C;",   # DEVANAGARI DIGIT SIX
    "f": "&#x10346;",  # GOTHIC LETTER FAIHU
    "g": "&#x1B9D;",   # SUNDANESE LETTER WA
    "h": "&#x10AC;",   # GEORGIAN CAPITAL LETTER NAR
    "i": "&#x10382;",  # UGARITIC LETTER GAMLA
    "j": "&#x1D4A5;",  # MATHEMATICAL SCRIPT CAPITAL J 
    "k": "&#x20AD;",   # KIP SIGN 
    "l": "&#x1D4DB;",  # MATHEMATICAL BOLD SCRIPT CAPITAL L 
    "m": "&#x0BF1;",   # TAMIL NUMBER ONE HUNDRED
    "n": "&#x0E81;",   # LAO LETTER KO
    "o": "&#x0C20;",   # TELUGU LETTER TTHA
    "p": "&#xA6E0;",   # BAMUM LETTER MBEN
    "q": "&#x1007;",   # MYANMAR LETTER JA
    "r": "&#x10342;",  # GOTHIC LETTER RAIDA
    "s": "&#x13A6;",   # CHEROKEE LETTER GA
    "t": "&#x1D366;",  # COUNTING ROD UNIT DIGIT SEVEN
    "u": "&#x07CE;",   # NKO LETTER U 
    "v": "&#x2A54;",   # DOUBLE LOGICAL OR
    "w": "&#x0E9C;",   # LAO LETTER PHO SUNG
    "x": "&#x0CF1;",   # KANNADA SIGN JIHVAMULIYA
    "y": "&#x10284;",  # LYCIAN LETTER G
    "z": "&#x2F04;"    # ANGXI RADICAL SECOND
}

m3 = {
    "a": "&#x10001;",  # LINEAR B SYLLABLE B038 E
    "b": "&#x2CD2;",   # COPTIC CAPITAL LETTER OLD COPTIC HEI 
    "c": "&#x1573;",   # CANADIAN SYLLABICS TYA
    "d": "&#x1D507;",  # MATHEMATICAL FRAKTUR CAPITAL D
    "e": "&#x0B3D;",   # TIBETAN LETTER DZA
    "f": "&#x2A0D;",   # FINITE PART INTEGRAL
    "g": "&#x01E4;",   # LATIN CAPITAL LETTER G WITH STROKE
    "h": "&#x12E0;",   # ETHIOPIC SYLLABLE ZHA
    "i": "&#x1703;",   # TAGALOG LETTER KA
    "j": "&#x3057;",   # HIRAGANA LETTER SI
    "k": "&#xA018;",   # CHEROKEE SMALL LETTER TSI
    "l": "&#x2827;",   # BRAILLE PATTERN DOTS-1236
    "m": "&#x0D28;",   # MALAYALAM LETTER NA
    "n": "&#x12AC;",   # ETHIOPIC SYLLABLE KEE
    "o": "&#x0DA7;",   # SINHALA LETTER ALPAPRAANA TTAYANNA
    "p": "&#xAA6D;",   # MYANMAR LETTER KHAMTI HA
    "q": "&#x10E2;",   # GEORGIAN LETTER TAR
    "r": "&#x053B;",   # ARMENIAN CAPITAL LETTER INI
    "s": "&#x1511;",   # CANADIAN SYLLABICS SHI
    "t": "&#x2020;",   # DAGGER
    "u": "&#x00B5;",   # MICRO SIGN 
    "v": "&#xAA9C;",   # TAI VIET LETTER LOW PO
    "w": "&#x0429;",   # CYRILLIC CAPITAL LETTER SHCHA 
    "x": "&#x102C4;",  # CARIAN LETTER NG
    "y": "&#x03AB;",   # GREEK CAPITAL LETTER UPSILON WITH DIALYTIKA 
    "z": "&#xA903;"    # KAYAH LI DIGIT THREE 
}

def to_struse(s1):
    s2 = ""
    for char in s1:
        c1 = char.lower()
        if c1 in m1:
            rn = random.randint(1, 3)
            if rn == 1:
                s2 += m1[c1]
            elif rn == 2:
                s2 += m2[c1]
            else:
                s2 += m3[c1]
        else:
            if char == " ":
                s2 += "&ensp; "
            else:
                s2 += c1
    return s2

if __name__ == "__main__":
    s1 = "rhymes with juice"
    for _ in range(4):
        print(to_struse(s1))

File: go/struse.go
// (c) 2026 : https://github.com/NMoroney + Gemini CLI
// MIT License
// https://github.com/NMoroney/Struse
// https://nmoroney.github.io/notes/2026/260423_struse/index.html
//

package main

import (
	"fmt"
	"math/rand"
	"strings"
	"time"
	"unicode"
)

var m1 = map[rune]string{
	'a': "&#x10300;", 'b': "&#x1041A;", 'c': "&#x1103;", 'd': "&#x1095;",
	'e': "&#x0F5B;", 'f': "&#x0CEF;", 'g': "&#x0D6D;", 'h': "&#x0A2E;",
	'i': "&#x1109;", 'j': "&#x10408;", 'k': "&#x051E;", 'l': "&#x10539;",
	'm': "&#x1328;", 'n': "&#x0E17;", 'o': "&#x07DB;", 'p': "&#x10502;",
	'q': "&#x09ED;", 'r': "&#x053B;", 's': "&#x0D17;", 't': "&#x10B5;",
	'u': "&#x053F;", 'v': "&#x13C9;", 'w': "&#x0E1F;", 'x': "&#x0514;",
	'y': "&#x080A;", 'z': "&#x10350;",
}

var m2 = map[rune]string{
	'a': "&#x1D70;", 'b': "&#x266D;", 'c': "&#x122D;", 'd': "&#x1D1CE;",
	'e': "&#x096C;", 'f': "&#x10346;", 'g': "&#x1B9D;", 'h': "&#x10AC;",
	'i': "&#x10382;", 'j': "&#x1D4A5;", 'k': "&#x20AD;", 'l': "&#x1D4DB;",
	'm': "&#x0BF1;", 'n': "&#x0E81;", 'o': "&#x0C20;", 'p': "&#xA6E0;",
	'q': "&#x1007;", 'r': "&#x10342;", 's': "&#x13A6;", 't': "&#x1D366;",
	'u': "&#x07CE;", 'v': "&#x2A54;", 'w': "&#x0E9C;", 'x': "&#x0CF1;",
	'y': "&#x10284;", 'z': "&#x2F04;",
}

var m3 = map[rune]string{
	'a': "&#x10001;", 'b': "&#x2CD2;", 'c': "&#x1573;", 'd': "&#x1D507;",
	'e': "&#x0B3D;", 'f': "&#x2A0D;", 'g': "&#x01E4;", 'h': "&#x12E0;",
	'i': "&#x1703;", 'j': "&#x3057;", 'k': "&#xA018;", 'l': "&#x2827;",
	'm': "&#x0D28;", 'n': "&#x12AC;", 'o': "&#x0DA7;", 'p': "&#xAA6D;",
	'q': "&#x10E2;", 'r': "&#x053B;", 's': "&#x1511;", 't': "&#x2020;",
	'u': "&#x00B5;", 'v': "&#xAA9C;", 'w': "&#x0429;", 'x': "&#x102C4;",
	'y': "&#x03AB;", 'z': "&#xA903;",
}

func init() {
	rand.Seed(time.Now().UnixNano())
}

// ToStruse performs Unicode inter-alphabet substitutions.
func ToStruse(s string) string {
	var builder strings.Builder
	for _, char := range s {
		cLower := unicode.ToLower(char)
		if _, exists := m1[cLower]; exists {
			rn := rand.Intn(3) + 1
			switch rn {
			case 1:
				builder.WriteString(m1[cLower])
			case 2:
				builder.WriteString(m2[cLower])
			case 3:
				builder.WriteString(m3[cLower])
			}
		} else {
			if char == ' ' {
				builder.WriteString("&ensp; ")
			} else {
				builder.WriteRune(cLower)
			}
		}
	}
	return builder.String()
}

func main() {
	input := "rhymes with juice"
	fmt.Println(ToStruse(input))
}
